[Doc] Update ST-Link doc following debugging chapter introduction

erwango opened the pull request:
Following #14000, a debugging section is introduced to describe properly debug probes and matching host tools. In this context it makes sense to remove existing doc/guides/tools/stlink-v21.rst and dispatch information in newly created files.
Remains some specific information, dedicated to OpenOcd/JLink usage with STLink which hardly fits in host-tool.rst or probes.rst, so create a new file stlink-v21.rst in debugging section. I'm open to other option to have that allow to have this information centralized in another place.

Based on #14000.

erwango committed:
doc/guides: debugging: Add ST-Link to probes

Add ST-Link chapter in probes page and reference in host-tool
openocd section.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

dbkinder reviewed (changes requested):
Looks OK, thanks.

erwango commented:
@dbkinder 
> Looks OK, thanks.

Well, can you approve then ? ;-)

dbkinder reviewed (approved):
Thought I had approved. Try again.

MaureenHelm commented on doc/guides/debugging/host-tools.rst:
```diff
@@ -54,6 +54,7 @@ These debug host tools are compatible with the following debug probes:
 
 - :ref:`opensda-daplink-onboard-debug-probe`
 - :ref:`jlink-external-debug-probe`
+- :ref:`stlink-v21-onboard-debug-probe`
```
Do you need to add a similar line to the J-Link host tools section?

erwango replied:
Yeah, will do

MaureenHelm commented on doc/guides/debugging/probes.rst:
```diff
@@ -137,6 +137,31 @@ As with all OpenSDA debug probes, the steps for programming the firmware are:
+It is compatible with the following host debug tools:
+
+- :ref:`openocd-debug-host-tools`
+- :ref:`jlink-debug-host-tools`
```
I think pyOCD recently added support for ST-Link

erwango replied:
Indeed, but not widely deployed so far. This is worth adding a comment anyway.

MaureenHelm commented on doc/guides/debugging/host-tools.rst:
```diff
@@ -88,3 +97,6 @@ Check if your SoC is listed in `pyOCD Supported Devices`_.
    https://github.com/mbedmicro/pyOCD/tree/master/pyocd/target/builtin
+
+.. _OpenOCD Windows:
+    http://gnutoolchains.com/arm-eabi/openocd/
```
Why here? What about the GNU MCU Eclipse mentioned in the "Getting OpenOCD" page?

erwango replied:
Seems this may not be the best way to get up to date openocd binaries (for windows at least). For windows, it redirects to http://www.freddiechopin.info/en/download/category/4-openocd with latest build having: 'build date: 2017-01-23'. (which is older than the oldest binary in http://gnutoolchains.com/arm-eabi/openocd/)

MaureenHelm commented on doc/guides/debugging/index.rst:
```diff
@@ -8,3 +8,4 @@ Debugging
 
    host-tools.rst
    probes.rst
+   stlink-v21.rst
```
Is this separate page really still necessary?

erwango replied:
Information provided in this page is not available anywhere else so it's worth keeping, IMHO. This being said I'm open to alternative locations.

MaureenHelm replied:
I think it can be worked into the debug probe page

erwango replied:
Ok, done!

MaureenHelm commented on doc/guides/debugging/probes.rst:
```diff
@@ -137,6 +137,97 @@ As with all OpenSDA debug probes, the steps for programming the firmware are:
+-------------
+
+Openocd is available by default on ST-Link and configure as default flash and
+debug tool, flash and debug could be done as follows:
```
Suggest rewording a little:
```
OpenOCD is available by default on ST-Link and configured as the default flash and
debug tool. Flash and debug can be done as follows:
```

erwango replied:
Txs, fixed

MaureenHelm commented on doc/guides/debugging/probes.rst:
```diff
@@ -137,6 +137,97 @@ As with all OpenSDA debug probes, the steps for programming the firmware are:
+Flash and debug with ST-Link
+============================
+
+Using Openocd
```
OpenOCD

erwango replied:
Fixed

erwango committed:
doc/guides/debugging: Add node for openocd support on windows

Add a note to mention how to get openocd working on windows.
This is taken from stlink-v21.rst file before its reformatting.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>
